src/Microsoft.AspNet.OData.Shared/Formatter/Serialization/SelectExpandNode.cs

```diff
@@ -126,13 +123,20 @@ public SelectExpandNode(SelectExpandClause selectExpandClause, IEdmStructuredTyp
                 HashSet<IEdmNavigationProperty> allNavigationProperties;
                 HashSet<IEdmAction> allActions;
                 HashSet<IEdmFunction> allFunctions;
+                IList<SelectItem> selectItems = new List<SelectItem>();
 
                 if (entityType != null)
                 {
                     allNavigationProperties = new HashSet<IEdmNavigationProperty>(entityType.NavigationProperties());
                     allActions = new HashSet<IEdmAction>(model.GetAvailableActions(entityType));
                     allFunctions = new HashSet<IEdmFunction>(model.GetAvailableFunctions(entityType));
                 }
+                else if (structuredType != null)
+                {
+                    allNavigationProperties = new HashSet<IEdmNavigationProperty>(structuredType.NavigationProperties());
+                    allActions = new HashSet<IEdmAction>();
+                    allFunctions = new HashSet<IEdmFunction>();
```

Comment:
why would these not be populated with available actions/functions, as in the entitytype case?

Reply:
Can operations be bound to complex types? EdmModelHelper method to get available functions only took in entity types. 

---
In reply to: [308488199](https://github.com/OData/WebApi/pull/1738#discussion_r308488199) [](ancestors = 308488199)

Reply:
Yes, protocol allows bounding operations to complex types. Would have to check to see what we do/do not support in the libraries.

---
In reply to: [308848817](https://github.com/OData/WebApi/pull/1738#discussion_r308848817) [](ancestors = 308848817,308488199)

Reply:
Checked in code and verified with Sam that we do not support bound actions/functions on complex type currently. 

---
In reply to: [308873937](https://github.com/OData/WebApi/pull/1738#discussion_r308873937) [](ancestors = 308873937,308848817,308488199)

Reply:
Can you open an issue to track adding support for bound operations to complex types?

---
In reply to: [326287637](https://github.com/OData/WebApi/pull/1738#discussion_r326287637) [](ancestors = 326287637,308873937,308848817,308488199)